manager/integration/tests/test_backing_image.py

```diff
@@ -511,6 +514,54 @@ def test_backing_image_auto_resync(client, volume_name):  # NOQA
     6. Wait for the file recovering automatically.
     7. Validate the volume.
     """
+    for bi_url in (BACKING_IMAGE_QCOW2_URL, BACKING_IMAGE_RAW_URL):
+        backing_image_auto_resync_test(bi_url, client, volume_name)
+
+
+def backing_image_auto_resync_test(bi_url, client, volume_name): # NOQA
+    # Step 1
+    create_backing_image_with_matching_url(
+              client, BACKING_IMAGE_NAME, bi_url)
+
+    # Step 2
+    volume = create_and_check_volume(
+                                     client, volume_name, 3,
+                                     str(BACKING_IMAGE_EXT4_SIZE),
+                                     BACKING_IMAGE_NAME)
+
+    # Step 3
+    lht_host_id = get_self_host_id()
+    volume.attach(hostId=lht_host_id)
+    volume = wait_for_volume_healthy(client, volume_name)
+    assert volume.backingImage == BACKING_IMAGE_NAME
+    assert volume.size == str(BACKING_IMAGE_EXT4_SIZE)
+
+    # Step 4
+    subprocess.check_output(['rm', '-rf', '/var/lib/longhorn/backing-images/'])
+
+    # Step 5
+    failed = False
+    for i in range(RETRY_BACKUP_COUNTS):
+        if failed is False:
+            backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+            for _, status in iter(backing_image.diskFileStatusMap.items()):
+                if status.state == "failed":
+                    failed = True
+        if failed is True:
+            break
+        time.sleep(0.1)
+    assert failed is True
+
```

Comment:
Could you make this as a reusable function instead? also, check which disk has that failed state instead of just checking one of them blindly. The disk should be the local one.

Reply:
Hi @innobead ,
By running `kubectl get backingimages -n longhorn-system <backing image name> -o json`, I can get  below information contain backing image status and their UUID, but according those information,  I can not find a way to identify which backing image located on which node. cc @shuo-wu  

For the reusable function I will start on that now.
```
"status": {
        "checksum": "bd79ab9e6d45abf4f3f0adf552a868074dd235c4698ce7258d521160e0ad79ffe555b94e7d4007add6e1a25f4526885eb25c53ce38f7d344dd4925b9f2cb5d3b",
        "diskDownloadProgressMap": null,
        "diskDownloadStateMap": null,
        "diskFileStatusMap": {
            "25bc5546-0e0d-4a73-a08d-f4196686351b": {
                "lastStateTransitionTime": "2022-09-22T02:37:46Z",
                "message": "",
                "progress": 100,
                "state": "ready"
            },
            "63183390-25e2-41f1-9bf8-d9448a2a5b38": {
                "lastStateTransitionTime": "2022-09-22T02:41:19Z",
                "message": "",
                "progress": 100,
                "state": "ready"
            },
            "a64656d4-0300-4892-b3f1-cb383275e1cd": {
                "lastStateTransitionTime": "2022-09-22T02:43:20Z",
                "message": "failed to re-calculate checksum after ready file being modified, will mark the file as failed: open /data/backing-images/test-vol-6db13dbb/backing: no such file or directory",
                "progress": 100,
                "state": "failed"
            }
        },

```

Reply:
Hi @innobead ,
I add `wait_for_backing_image_having_fail` in common.py for review, thank you

Reply:
Hi @innobead , 
I added `wait_for_backing_image_fail(client, backing_img_name, disk_uuid)` in common.py to wait specific disk_uuid's backing image state become fail, thank you